static/js/components/widgets/SiteContentEmbedField.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useState, useCallback, SyntheticEvent } from "react"
+import { useSelector } from "react-redux"
+import { useRequest } from "redux-query-react"
+
+import { useWebsite } from "../../context/Website"
+import {
+  websiteContentListingRequest,
+  WebsiteContentListingResponse
+} from "../../query-configs/websites"
+import { getWebsiteContentListingCursor } from "../../selectors/websites"
+import { ContentListingParams } from "../../types/websites"
+import { useDebouncedState } from "../../hooks/state"
+
+interface Props {
+  insertEmbed: (id: string) => void
+  attach: string
+}
+
+export default function SiteContentEmbedField(props: Props): JSX.Element {
+  const { insertEmbed, attach } = props
+
+  // filterInput is to store user input and is updated synchronously
+  // so that the UI stays responsive
+  const [filterInput, setFilterInput] = useState("")
+  // filter, by contrast, is set by the setFilterDebounced function
+  // to cut down on extraneous requests. filter is set as a param in
+  // the listingParams object below, so updating the value causes the
+  // websiteContentListingRequest to be re-run, so we debounce for
+  // less silliness!
+  const [filter, setFilterDebounced] = useDebouncedState("", 300)
+
+  const onChangeFilter = useCallback(
+    (event: SyntheticEvent<HTMLInputElement>) => {
+      const newFilter = event.currentTarget.value
+      setFilterInput(newFilter)
+      setFilterDebounced(newFilter)
+    },
+    [setFilterDebounced]
+  )
+
+  const website = useWebsite()
+  const listingParams: ContentListingParams = {
+    name:   website.name,
+    type:   attach,
+    search: filter,
+    offset: 0
+  }
+
+  useRequest(websiteContentListingRequest(listingParams, false, false))
+
+  const listing: WebsiteContentListingResponse = useSelector(
+    getWebsiteContentListingCursor
+  )(listingParams)
+
+  return (
+    <div className="my-2 content-embed-widget">
+      <label>Resources</label>
+      <input
+        placeholder="filter"
+        type="text"
+        onChange={onChangeFilter}
+        value={filterInput}
+        className="form-control filter-input"
+      />
+      <div className="content-list">
+        {listing.results ?
+          listing.results.map(item => (
+            <div
```

Comment:
Again, I'm not sure if this is outside of the scope of this PR but it would be nice to see some additional info here.  At least `filetype` would be nice.  For example with transcripts, there might be 3 or 4 different files labeled with the same title but a different `filetype`.

Reply:
maybe I'll do a switch on the filetype to display a different icon, to keep it small and lightweight

Reply:
I think the problem we start to run into then is that the `filetype` field is defined in the website start config file, so if we add a display of that field here then we're sort of coupling the UI field here with a specific type of course config. I'm not sure the best way around that - we could for instance add some more config options that define how a metadata field from the 'resource' content items should be grabbed, but that could get kind of complicated too.

Anyhow, not sure what you think about what would be the best way forward there

Reply:
I think it's certainly outside the scope of this PR but definitely worth a discussion with the group.  I don't think just the title is enough context overall, especially when a lot of resources start piling up.